generators/generator-base-blueprint.js

```diff
@@ -84,51 +90,187 @@ module.exports = class extends BaseGenerator {
+                otherModules.push(...blueprints);
             }
 
-            blueprints.forEach(blueprint => {
-                let bpOptions = {
-                    ...this.options,
-                    configOptions: this.configOptions,
-                };
-                if (extraOptions) {
-                    bpOptions = { ...bpOptions, ...extraOptions };
+            this.jhipsterConfig.blueprints = blueprints;
+            this.jhipsterConfig.otherModules = otherModules;
+
+            if (!this.options.skipChecks) {
+                const namespaces = blueprints.map(bp => jhipsterUtils.packageNameToNamespace(bp.name));
+                // Verify if the blueprints has been registered.
+                const missing = namespaces.filter(namespace => !this.env.isPackageRegistered(namespace));
+                if (missing && missing.length > 0) {
+                    this.error(`Some blueprints were not found ${missing}, you should install them manually`);
                 }
-                const blueprintGenerator = this.composeBlueprint(blueprint.name, subGen, bpOptions);
+            }
+        }
+
+        const blueprints = this.jhipsterConfig.blueprints;
+        if (blueprints && blueprints.length > 0) {
+            blueprints.forEach(blueprint => {
+                const blueprintGenerator = this.composeBlueprint(blueprint.name, subGen, extraOptions);
                 // If the blueprints sets sbsBlueprint property, then don't ignore the normal workflow.
-                if (!useBlueprints && blueprintGenerator && !blueprintGenerator.sbsBlueprint) {
+                if (blueprintGenerator && !blueprintGenerator.sbsBlueprint) {
                     useBlueprints = true;
                 }
             });
         }
         return useBlueprints;
     }
+
+    /**
+     * Compose external blueprint module
+     * @param {string} blueprint - name of the blueprint
+     * @param {string} subGen - sub generator
+     * @param {any} options - options to pass to blueprint generator
+     */
+    composeBlueprint(blueprint, subGen, extraOptions = {}) {
+        blueprint = jhipsterUtils.normalizeBlueprintName(blueprint);
+        if (!this.configOptions.skipChecks && !this.options.skipChecks) {
+            this.checkBlueprint(blueprint);
+        }
+
+        const generatorName = jhipsterUtils.packageNameToNamespace(blueprint);
+        const generatorNamespace = `${generatorName}:${subGen}`;
+        if (!this.env.isPackageRegistered(generatorName)) {
+            this.env.lookup({ filterPaths: true, packagePatterns: blueprint });
+        }
+        if (!this.env.get(generatorNamespace)) {
+            this.debug(
+                `No blueprint found for blueprint ${chalk.yellow(blueprint)} and ${chalk.yellow(
+                    subGen
+                )} subgenerator: falling back to default generator`
+            );
+            return false;
```

Comment:
There's an issue with what's returned: I see a boolean here (perhaps it should be `undefined`), but also an object l.190.